core/hooks/LayoutProps.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+
+import {getLayoutProps, getNonLayoutProps} from '@xh/hoist/utils/react';
+
+/**
+ * This hook provides support for flexbox related styles that are set as top-level properties
+ * on a component.
+ *
+ * The following properties will be supported:
+ *     margin, marginTop, marginRight, marginBottom, marginLeft,
+ *     padding, paddingTop, paddingRight, paddingBottom, paddingLeft,
+ *     height, minHeight, maxHeight, width, minWidth, maxWidth,
+ *     flex, flexBasis, flexDirection, flexGrow, flexShrink, flexWrap,
+ *     alignItems, alignSelf, alignContent, justifyContent,
+ *     overflow, overflowX, overflowY,
+ *     top, left, position, display
+ *
+ * Use of this hook indicates that the Component will respect and respond to these properties.
+ * Components will typically delegate this responsibility to a child component that also
+ * implements LayoutSupport. `Box` is typically the Component that is ultimately rendered
+ * and will handle this by outputting a div with appropriate styles.
+ *
+ * @param {Array} props
+ * @returns {Object[]} - Array containing a bundle of layout props, and a bundle of all non-layout props.
+ */
+export function useLayoutProps(props) {
+    // Consider caching this comp if props haven't changed.
+    return [getLayoutProps(props), getNonLayoutProps(props)];
```

Comment:
I think this is a better candidate for a hook than the class names because it is such a core cross-cutting bit of Component functionality, and I could see it relying on the component lifecycle, or caching, or some specialized logging/correctness checking that happens  only on mount?

If I understand useMemo correctly, we would actually need to specify the full lot of the parameters in the dependencies array, so I am not sure that is actually a good use at the moment -- although I wonder if it might change if we ever did more complicated layout computations.

I could actually imagine an optimization that just compares the provided props *that are not functions* against each other (for example)..... Or an option on the hook that declares itself for
"static" layout only .